static/src/js/components/CollectionSelector/CollectionSelector.jsx

```diff
@@ -0,0 +1,425 @@
+              directDistributionInformation
+            }
+          })
+
+          callback(options)
+        }
+      })
+    }
+  }, 1000), [])
+
+  const handleAvailableSearchChange = (e) => {
+    const inputValue = e.target.value
+    setSearchAvailable(inputValue)
+    loadOptions(inputValue, setAvailable)
+  }
+
+  const popover = (item) => {
+    const {
+      directDistributionInformation,
+      entryTitle,
+      shortName
+    } = item
+    const { s3BucketAndObjectPrefixNames } = directDistributionInformation || {}
+
+    return (
+      <Popover>
+        <Popover.Body>
+          {shortName}
+          {' '}
+          |
+          {' '}
+          {entryTitle}
+          {
+            directDistributionInformation && (
+              <span>
+                {
+                  s3BucketAndObjectPrefixNames?.map(
+                    (prefix) => (
+                      <div
+                        key={prefix}
+                      >
+                        <strong>S3 Prefix: </strong>
+                        {' '}
+                        {prefix}
+                      </div>
+                    )
+                  )
+                }
+              </span>
+            )
+          }
+        </Popover.Body>
+      </Popover>
+    )
+  }
+
+  return (
+    <Row className="mb-4">
+      <Col>
+        <div className="border rounded p-3">
+          <h5 className="text-center mb-3">Available Collections</h5>
+          <Form.Control
+            className="mb-3"
+            onChange={handleAvailableSearchChange}
+            placeholder="Search Available..."
+            type="text"
+            value={searchAvailable}
+          />
+          <div className="collection-selector__list-group  d-block w-100 overflow-y-scroll border rounded">
+            <ul className="list-unstyled h-100">
+              <For each={available}>
+                {
+                  (
+                    item
+                  ) => {
+                    const {
+                      conceptId,
+                      directDistributionInformation,
+                      entryTitle,
+                      provider,
+                      shortName
+                    } = item
+
+                    const title = `${shortName} | ${entryTitle}`
+
+                    return (
+                      <OverlayTrigger
+                        key={conceptId}
+                        overlay={popover(item)}
+                        placement="top"
+                      >
+                        <li
+                          aria-hidden="true"
+                          aria-label={title}
+                          className={
+                            classNames(
+                              'collection-selector__list-group-item d-flex justify-content-between align-items-center px-3 py-2',
+                              {
+                                'collection-selector__list-group-item-secondary': selected.some(
```

Comment:
I think the primary and secondary classes here should be written as modifiers of the base `collection-selector__list-group-item` and not as elements. Take a look at the BEM methodology documentation. Happy to help if you need clarification.